hi-authen/src/com/viettel/authen/db/daoImpl/UserDaoImpl.java

```diff
@@ -5,28 +5,111 @@
  */
 package com.viettel.authen.db.daoImpl;
 
-import com.viettel.authen.db.dao.UserDao;
-import com.viettel.authen.run.StartApp;
 import java.sql.SQLException;
 import java.util.ArrayList;
 import java.util.Date;
+import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 
+import com.google.gson.Gson;
+import com.viettel.authen.db.dao.UserDao;
+import com.viettel.authen.run.StartApp;
+import org.apache.log4j.*;
+
 /**
  *
  * @author HienDM
  */
 
 public class UserDaoImpl implements UserDao {
+	private static final int UPDATE_TYPE = 1;
+	private static final int DELETE_TYPE = 0;
+	private static final Logger log = Logger.getLogger(UserDaoImpl.class.getName());
+	private static Gson gson = new Gson();
 
 	public List<Map> getAllUser() throws SQLException {
 		return StartApp.database.queryData("select * from sm_user");
 	}
 
+	// This is not an optimize way to do this, but we need a work around now so
+	// please forgive me!
+	/**
+	 * 
+	 * @param userName: User name that you want to update the information to hicache
+	 */
+	private void updateHiCacheUserCredential(Map resultMap, Map<Object, Object> user, int userId) {
+		log.debug("TOGREP | User information updated: " + resultMap);
+
+		// Set user information
+		StartApp.hicache.useSpace("authen");
+		StartApp.hicache.setStoreAttribute("credentials_id", "" + userId, user.get("user_name"));
+		StartApp.hicache.setStoreAttribute("credentials", user.get("user_name").toString(), new Gson().toJson(user));
+	}
+
+	private void updateHiCachePasswordExpiry(Map resultMap, Map<Object, Object> user) {
+		StartApp.hicache.useSpace("authen");
+		StartApp.hicache.setStoreAttribute("expiry_password", user.get("user_name").toString(),
+				resultMap.get("password_expiry_date").toString());
+	}
+
+	private void updateHiCache(int userId, int type) {
+		switch (type) {
```

Comment:
@ngocdon0127 Please use this function everytime IP list need to be updated to cache.